.github/workflows/benchmark.yml

```diff
@@ -19,32 +19,107 @@ on:
         description: 'Number of job splits'
         required: true
         default: '1'
+      run-tcp-ds-query-benchmark:
+        description: 'Run TPCDSQueryBenchmark: true or false'
+        required: false
+        default: 'false'
 
 jobs:
   matrix-gen:
     name: Generate matrix for job splits
     runs-on: ubuntu-20.04
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      class: ${{ steps.set-matrix.outputs.class }}
+      split: ${{ steps.set-matrix.outputs.split }}
+      num-split: ${{ steps.set-matrix.outputs.num-split }}
     env:
+      SPARK_BENCHMARK_CLASS: ${{ github.event.inputs.class }}
       SPARK_BENCHMARK_NUM_SPLITS: ${{ github.event.inputs.num-splits }}
     steps:
     - name: Generate matrix
       id: set-matrix
-      run: echo "::set-output name=matrix::["`seq -s, 1 $SPARK_BENCHMARK_NUM_SPLITS`"]"
+      run: |
+        echo "::set-output name=class::['"$SPARK_BENCHMARK_CLASS"']"
+        echo "::set-output name=split::["`seq -s, 1 $SPARK_BENCHMARK_NUM_SPLITS`"]"
+        echo "::set-output name=num-split::['"$SPARK_BENCHMARK_NUM_SPLITS"']"
+
+  generate-TCP-DS-dataset:
```

Comment:
Can you leave a cross comment in build_and_test.yml so we don't forgot to update together?

Reply:
thank you for your comment! Addressed in an additional commit